src/main/java/com/flipkart/zjsonpatch/InternalUtils.java

```diff
@@ -0,0 +1,64 @@
+package com.flipkart.zjsonpatch;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+class InternalUtils {
+
+    static List<JsonNode> toList(ArrayNode input) {
+        List<JsonNode> toReturn = new ArrayList<JsonNode>();
+        int size = input.size();
+        for (int i = 0; i < size; i++) {
+            toReturn.add(input.get(i));
+        }
+        return toReturn;
+    }
+
+    private static List<JsonNode> reverse(List<JsonNode> list) {
+        List<JsonNode> toReturn = new LinkedList<JsonNode>();
+        for (int i = list.size() - 1; i >= 0; i--) {
```

Comment:
Would probably be better to forward-iterate the list (the interface makes no guarantees about time complexity on indexed retrieval/update) and build up an ArrayList (which does).

Reply:
We could use Collections.reverse, which guarantees linear time complexity instead of this method

Reply:
That requires a clone first, not sure it'll end up being any faster...

Reply:
Inside InternalUtils.longestCommonSubsequence, we could use Collections.reverse ( it does in-place reversal ) before the return statement. 
By your last comment, did you mean that sir?

Reply:
It's possible, but may be slightly less readable. Bottom line, we don't have a benchmark suite yet, so the whole discussion is kind of moot :-)

Reply:
I am using @cyber-front test-suite ( shared sometime back with both of us ) for benchmarking where it creates 65,536 random objects and performed a series of 256 updates on each.